# 1 Introduction and clinical context

# 2 Defintions methods and hypotheses

# 3 Import of data and libraries

## 3.1 Import of libraries

In [8]:
#Importing basic data analyst libraries
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

#Importing libraries for SQL querrying
import duckdb 

##3.2 Importing tables

I decided to import all the tables relevant for my analysis at this point. I will treat them as a database and later on i will look at them in detail. Now I will just import them one by one whereas querrying and cleaning comes a bit later in separate stages. I chose this approach because i find it more logically organised compared to other solutions.

###3.2.1 Importing confirmed cases table

In [12]:
df_cases = pd.read_csv('data/raw/confirmed_global.csv')

Verifying import

In [14]:
print(df_cases.head())

  Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  4/1/21  4/2/21  4/3/21  4/4/21  \
0        0        0        0        0  ...   56517   56572   56595   56676   
1        0        0        0        0  ...  125506  125842  126183  126531   
2        0        0        0        0  ...  117304  117429  117524  117622   
3        0        0        0        0  ...   12053   12115   12174   12231   
4        0        0        0        0  ...   22399   22467   22579   22631   

   4/5/21  4/6/21  4/7/21  4/8/21  4/9/21  4/10/21  
0   56717   56779   56873   56943

##### ---Cases dataset imported successfully---

### 3.2.2 Importing death cases

In [17]:
df_deaths = pd.read_csv('data/raw/deaths_global.csv')

Verifying import

In [23]:
print(df_deaths.head())

  Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  4/1/21  4/2/21  4/3/21  4/4/21  \
0        0        0        0        0  ...    2489    2495    2496    2497   
1        0        0        0        0  ...    2241    2247    2256    2265   
2        0        0        0        0  ...    3096    3099    3102    3105   
3        0        0        0        0  ...     115     116     117     117   
4        0        0        0        0  ...     538     538     540     542   

   4/5/21  4/6/21  4/7/21  4/8/21  4/9/21  4/10/21  
0    2508    2512    2512    2516

##### ----Deaths dataset importet successfully----

### 3.2.3 Importing recovered cases dataset

In [27]:
df_recovered = pd.read_csv('data/raw/recovered_global.csv')

In [29]:
print(df_recovered.head())

  Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  4/1/21  4/2/21  4/3/21  4/4/21  \
0        0        0        0        0  ...   51788   51798   51802   51885   
1        0        0        0        0  ...   91875   92500   93173   93842   
2        0        0        0        0  ...   81632   81729   81813   81896   
3        0        0        0        0  ...   11365   11401   11428   11474   
4        0        0        0        0  ...   20508   20867   20871   20879   

   4/5/21  4/6/21  4/7/21  4/8/21  4/9/21  4/10/21  
0   51902   51928   51940   51956

##### ----Recovered dataset importet successfully!----

# 4 Inspecting cleaning and preparing datasets for explanatory data analysis

In this sections I will get acquianted with dataset, melt it, deal with missings and prepare it for the data analysis

## 4.1 Inspecting cleaning and preparing dataset with cases

### 4.1.1 Inspecting dataset with cases

In [36]:
print(df_cases.head())

  Province/State Country/Region       Lat       Long  1/22/20  1/23/20  \
0            NaN    Afghanistan  33.93911  67.709953        0        0   
1            NaN        Albania  41.15330  20.168300        0        0   
2            NaN        Algeria  28.03390   1.659600        0        0   
3            NaN        Andorra  42.50630   1.521800        0        0   
4            NaN         Angola -11.20270  17.873900        0        0   

   1/24/20  1/25/20  1/26/20  1/27/20  ...  4/1/21  4/2/21  4/3/21  4/4/21  \
0        0        0        0        0  ...   56517   56572   56595   56676   
1        0        0        0        0  ...  125506  125842  126183  126531   
2        0        0        0        0  ...  117304  117429  117524  117622   
3        0        0        0        0  ...   12053   12115   12174   12231   
4        0        0        0        0  ...   22399   22467   22579   22631   

   4/5/21  4/6/21  4/7/21  4/8/21  4/9/21  4/10/21  
0   56717   56779   56873   56943

In [38]:
print(df_cases.shape)

(274, 449)


Shape of the dataset is 274 rows and 449 columns. 

From the first glance I see identifier columns such as province/state, Country/Relogion, Latitude and longitude. There are 449 columns where majority corresponds to individual dates within given period of time. To confirm that only date columns follow 'Long', and that there are no unexpected columns in between, I will iterate through the column names and count the columns to crosscheck whether i itterate through columns correctly.

In [42]:
count = 0
for col in df_cases:
    print(col)
    count = count + 1
    
print("the column count is: ", count)

Province/State
Country/Region
Lat
Long
1/22/20
1/23/20
1/24/20
1/25/20
1/26/20
1/27/20
1/28/20
1/29/20
1/30/20
1/31/20
2/1/20
2/2/20
2/3/20
2/4/20
2/5/20
2/6/20
2/7/20
2/8/20
2/9/20
2/10/20
2/11/20
2/12/20
2/13/20
2/14/20
2/15/20
2/16/20
2/17/20
2/18/20
2/19/20
2/20/20
2/21/20
2/22/20
2/23/20
2/24/20
2/25/20
2/26/20
2/27/20
2/28/20
2/29/20
3/1/20
3/2/20
3/3/20
3/4/20
3/5/20
3/6/20
3/7/20
3/8/20
3/9/20
3/10/20
3/11/20
3/12/20
3/13/20
3/14/20
3/15/20
3/16/20
3/17/20
3/18/20
3/19/20
3/20/20
3/21/20
3/22/20
3/23/20
3/24/20
3/25/20
3/26/20
3/27/20
3/28/20
3/29/20
3/30/20
3/31/20
4/1/20
4/2/20
4/3/20
4/4/20
4/5/20
4/6/20
4/7/20
4/8/20
4/9/20
4/10/20
4/11/20
4/12/20
4/13/20
4/14/20
4/15/20
4/16/20
4/17/20
4/18/20
4/19/20
4/20/20
4/21/20
4/22/20
4/23/20
4/24/20
4/25/20
4/26/20
4/27/20
4/28/20
4/29/20
4/30/20
5/1/20
5/2/20
5/3/20
5/4/20
5/5/20
5/6/20
5/7/20
5/8/20
5/9/20
5/10/20
5/11/20
5/12/20
5/13/20
5/14/20
5/15/20
5/16/20
5/17/20
5/18/20
5/19/20
5/20/20
5/21/20
5/22/20
5/23/20
5/24/20
5/25/

All the respective columns  after 'Long' seem to represent individual dates and column count from the loop is in agreement with the column count from df_cases.head(). Next I need to check data types of individual columns

In [50]:
count_integer = 0
count_other_datatypes = 0
for col in df_cases:
    print(df_cases[col].dtype)
    if df_cases[col].dtype == int:
        count_integer = count_integer +1
    else:
        count_other_datatypes = count_other_datatypes + 1

print('Integer count is:', count_integer, '\n Other data types count is:', count_other_datatypes)
if count_integer == 445:
    print('All date-columns are integers!')

object
object
float64
float64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int64
int6

Loop confirmed that all date columns are formated as an integer datatype.

In [48]:
print(df_cases.describe)

<bound method NDFrame.describe of     Province/State      Country/Region        Lat        Long  1/22/20  \
0              NaN         Afghanistan  33.939110   67.709953        0   
1              NaN             Albania  41.153300   20.168300        0   
2              NaN             Algeria  28.033900    1.659600        0   
3              NaN             Andorra  42.506300    1.521800        0   
4              NaN              Angola -11.202700   17.873900        0   
..             ...                 ...        ...         ...      ...   
269            NaN             Vietnam  14.058324  108.277199        0   
270            NaN  West Bank and Gaza  31.952200   35.233200        0   
271            NaN               Yemen  15.552727   48.516388        0   
272            NaN              Zambia -13.133897   27.849332        0   
273            NaN            Zimbabwe -19.015438   29.154857        0   

     1/23/20  1/24/20  1/25/20  1/26/20  1/27/20  ...  4/1/21  4/2/21  4/3/21

Describe command gives me an overview over the dataset. There are a lot of missing values in province/state. I have to investigate this variable closer. Country/region column will likely be the variable of interest and will serve us as identifier, Whereas latitute and longitude are not interesting forour analysis. Even though some viruses are known to spread better in colder or warmer climates, Cross-country comparisons in cases between the countries are not feasible because we don´t have data on people tested and countries were very different in test measures and  test equipment. In this dataset we lack the data for it. I will not check dataset with .describe(). Quick overview shows that there were 0 cases in a begining whereas number of cases began to increase and fluctuate over time.